src/net/i2p_address.h

```diff
@@ -104,7 +103,7 @@ namespace net
         const char* host_str() const noexcept { return host_; }
 
         //! \return Port value or `0` if unspecified.
-        std::uint16_t port() const noexcept { return port_; }
+        std::uint16_t port() const noexcept { return 1; }
```

Comment:
A comment explaining why this doesn't default to/return zero could be helpful (also the doc is somewhat outdated).

Reply:
Updated comment.